core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -720,6 +730,7 @@ void CudaInternal::finalize() {
   if (this == &singleton()) {
     cudaFreeHost(constantMemHostStaging);
     cudaEventDestroy(constantMemReusable);
+    cudaStreamDestroy(cuda_get_deep_copy_stream());
```

Comment:
Note that now we pay for a stream create even if we never transferred data between host and device.  I don't think that's a real issue, just making sure you see it.

Reply:
where is the destruction of the Cuda instance?

Reply:
At the exit of the program

Reply:
Can't we just call `finalize` on the `static` execution space instance we create?

Reply:
https://github.com/kokkos/kokkos/blob/0760c201443cce531016188cc9d7fbb53fe426f9/core/src/impl/Kokkos_SharedAlloc.cpp#L234-L246

Reply:
Are you talking about https://github.com/kokkos/kokkos/blob/1fb0c284d458c75370094921d9f202c287502325/core/src/impl/Kokkos_SharedAlloc.cpp#L238-L246? Decrementing the counter itself should not be a problem in itself, or?

We set `g_is_initialized` after calling `finalize_all_spaces` so we should not run into this assertion when we also finalize the `static` execution space used in` DualView` here or am I missing something?